InfinispanAPIObject.adoc

```diff
@@ -0,0 +1,36 @@
+== Infinispan API
+
+=== Motivation
+
+Being able to offer a simple API to create clustered caches and improve API usability
```

Comment:
Well, since competitors offer this kind of super class and I discovered it because I've playing with their demos, I really think it does belong in our code